### ISSUE TITLE
No se puede crear nuevo usuario 

### ISSUE THREAD

**Issue by tuxrick** [conekta/conekta-php#111]
Parse error: syntax error, unexpected T_OBJECT_OPERATOR in /home/obmexicz/public_html/lib/lib/Conekta/ConektaResource.php on line 206

<?php
require_once("lib/lib/Conekta.php");
//ProdKey
\Conekta\Conekta::setApiKey("key_#######");
\Conekta\Conekta::setApiVersion("2.0.0");

$token = $_POST["token"];
$name = $_POST["name"];
$email = $_POST["email"];
$quantity = $_POST["quantity"];
try {
	
	  $customer = \Conekta\Customer::create(
	    array(
	      "name" => $name,
	      "email" => $email,
	      "phone" => "+52181818181",
	      "payment_sources" => array(
	        array(
	            "type" => "card",
	            "token_id" => $token
	        )
	      )//payment_sources
	    )//customer
	  );
	//print_r($customer->name);
	//print_r($customer->name);
	//print_r($customer->email);	
	//print_r($customer->id);
} catch (\Conekta\ProccessingError $error){
		$error_final = $error->getMessage();
	   header("Location: https://obmex.org/index.html?error=".$error_final);
} catch (\Conekta\ParameterValidationError $error){
		$error_final = $error->getMessage();
	   header("Location: https://obmex.org/index.html?error=".$error_final);
} catch (\Conekta\Handler $error){
		$error_final = $error->getMessage();
	   header("Location: https://obmex.org/index.html?error=".$error_final);
}
?>

**Comment by chuckrincon** [comment]
¿@tuxrick Puedes adjuntar más detalles de cómo reproducir el escenario?  
`(variables de configuración, el código para crear el usuario, etc)`.

**Comment by tuxrick** [comment]
index.html

<!DOCTYPE html>
<html>
<head>
	<meta charset="utf-8"/>
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" type="text/css" href="css/style.css"/>
	<link rel="stylesheet" type="text/css" href="css/bootstrap.css"/>
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">

	
	<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js"></script>
	<script type="text/javascript" src="https://cdn.conekta.io/js/latest/conekta.js"></script>
	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
	<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
	
	
	<title>Operación Bendición México</title>
</head>
<body onload="fncError()">
	<nav class="">
		<div class="container_nav">
			<div class="logo">
				<img src="img/logo_obm-02.png" alt="logo_trebol_bit" height="65px">
				<button type="button" class="dona_button btn btn-info btn-lg btn-primary" data-toggle="modal" data-target="#myModal" style="margin-bottom: 25px;">Donar ahora</button>
			</div>
		</div>
	</nav>
	<div class="principal_image">
		<img style="width: 100%;" src="img/construct_obmex.jpg">
	</div>
	<div class="row">
		<div class="reconstruyendo_logo col-md-12">
			<img class="second_logo" src="img/recons_logo-02.png">
		</div>
	</div>
	<div class="row">	
		<div class="infos">
			<div class="text_bold" style="margin-bottom: 25px;">
				<span class="">Operación Bendición México se está actualizando para seguir construyendo sueños y una nueva sociedad.</span>
			</div>
			<div class="text_light" style="margin-bottom: 25px;">
				<span class="">Si deseas hacer tu donación puedes hacerlo en cuestión de minutos y apoya a las comunidades de Puebla, Oaxaca, Estado de México y Chiapas</span>
			</div>	
		</div>
		<div>
			<button type="button" class="dona_button btn btn-info btn-lg btn-primary" data-toggle="modal" data-target="#myModal" style="margin-bottom: 25px;">Donar ahora</button>
		</div>	
	</div>				





<div id="myModal" class="modal fade" role="dialog">
  <div class="modal-dialog">

    
    <div class="modal-content">
      <div class="modal-header">
	    <h4 class="modal-title">Donar ahora</h4>
        <button type="button" class="close" data-dismiss="modal">&times;</button>
      </div>
      <div class="modal-body">
        <form action="createCustomer.php" method="POST" id="card-form">
		  <span class="card-errors"></span>
		  <div>
		    <label>
		      <span class="form_text">Nombre del tarjetahabiente</span>
		      <input type="text" size="20" data-conekta="card[name]" name="name" id="name">
		    </label>
		  </div>
		  <div>
		    <label>
		      <span class="form_text">Correo electrónico</span>
		      <input type="text" size="20" name="email" id="email">
		    </label>
		  </div>
		  <div>
		    <label>
		      <span class="form_text">Cantidad a donar (Pesos)</span>
		      <input type="number" size="20" name="quantity" id="quantity">
		    </label>
		  </div>		  
		  <div>
		    <label>
		      <span class="form_text">Número de tarjeta de crédito</span>
		      <input type="text" size="20" data-conekta="card[number]">
		    </label>
		  </div>
		  <div>
		    <label>
		      <span class="form_text">CVC</span>
		      <input type="text" size="4" data-conekta="card[cvc]">
		    </label>
		  </div>
		  <div>
		    <label>
		      <span class="form_text">Fecha de expiración (MM/AAAA)</span>
		      <select type="text" data-conekta="card[exp_month]">
			      <option value="01">01</option>
			      <option value="02">02</option>
			      <option value="03">03</option>
			      <option value="04">04</option>
			      <option value="05">05</option>
			      <option value="06">06</option>
			      <option value="07">07</option>
			      <option value="08">08</option>
			      <option value="09">09</option>
			      <option value="10">10</option>
			      <option value="11">11</option>
			      <option value="12">12</option> 
		      </select>    
		    </label>
		    <span>/</span>
		    <select type="text" data-conekta="card[exp_year]">
				<option value="2017">2017</option>
			    <option value="2018">2018</option>
			    <option value="2019">2019</option>
			    <option value="2020">2020</option>
			    <option value="2021">2021</option>
			    <option value="2022">2022</option>
			    <option value="2023">2023</option>
			    <option value="2024">2024</option>
			    <option value="2025">2025</option>
			    <option value="2026">2026</option>
			    <option value="2027">2027</option>
			    <option value="2028">2028</option>
			    <option value="2029">2029</option>
			    <option value="2030">2030</option>
			    <option value="2031">2031</option>
			    <option value="2032">2032</option>
			    <option value="2033">2033</option>
			    <option value="2034">2034</option>
			    <option value="2035">2035</option>
		    </select>    
		  </div>
		  
		  <button class="dona_btn_modal" type="submit">Donar</button>
		</form>
      </div>
      <div class="modal-footer">
        <button type="button" class="btn btn-default" data-dismiss="modal">Cerrar</button>
      </div>
    </div>

  </div>
</div>

<script type="text/javascript" >

  Conekta.setPublishableKey('key_I9JPLivQHjMso69q5EGPKwg');

  var conektaSuccessResponseHandler = function(token){
    var correctValues=validaCampos();
     if (correctValues==true) {
     		//alert('ok');
     			var $form = $("#card-form");
	    //Inserta el token_id en la forma para que se envíe al servidor
	     $form.append($('<input type="hidden" name="token" id="conektaTokenId">').val(token.id));
	     //alert(token.id);
	     console.log(token.id);
     	$form.get(0).submit(); //Hace submit	     	
     }
     else{
     	if (correctValues==1) {
     		alert("Nombre Inválido");
     	}
     	else if(correctValues==2){
     		alert("Formato de e-mail Invalido");
     	}
     	else if(correctValues==3){
     		alert("Solo puedes donar más de 300 pesos y menos de 5000");
     	}

     	//alert(correctValues);
     }     
  };
  var conektaErrorResponseHandler = function(response) {
    var $form = $("#card-form");
    $form.find(".card-errors").text(response.message_to_purchaser);
    $form.find("button").prop("disabled", false);
  };

  //jQuery para que genere el token después de dar click en submit
  $(function () {
    $("#card-form").submit(function(event) {
      var $form = $(this);
      // Previene hacer submit más de una vez
      $form.find("button").prop("disabled", true);
      Conekta.token.create($form, conektaSuccessResponseHandler, conektaErrorResponseHandler);
      return false;
    });
  });
function validaCampos(){
//	console.log('llega');
var vname=$('#name').val();
var vemail=$('#email').val();
var vquantity=$('#quantity').val();
var cat_donat=parseFloat(vquantity);
var name_len=vname.length;
//console.log(vname, vemail, vquantity);
//console.log(name_len+'   lenght nombre');
	if (name_len!="") {
		var res = validateEmail(vemail);
		console.log(res);
		if (res!=false) {
			if(cat_donat>299 && cat_donat<5000){
				return true
			}
			else{
				return 3;
			}
		}
		else{
			return 2;
		}
	}
	else{
		return 1;
	}
//return false;
}
function validateEmail(email) {
	var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
	return re.test(email);
}
function fncError(){
	var c = getParameterByName('error');
	if (c!=null) {
		alert('Error: '+c);	
	}
	
}
function getParameterByName(name, url) {
    if (!url) url = window.location.href;
    name = name.replace(/[\[\]]/g, "\\$&");
    var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
        results = regex.exec(url);
    if (!results) return null;
    if (!results[2]) return '';
    return decodeURIComponent(results[2].replace(/\+/g, " "));
}
</script>

</body>
</html>

////Create Order PHP 

<!DOCTYPE html>
<?php


$id = $_POST["id"];
$quantity = $_POST["quantity"];

require_once("lib/lib/Conekta.php");
//ProdKey
\Conekta\Conekta::setApiKey("key_9DgJKqBokDgDhhzbhHhemw");
\Conekta\Conekta::setApiVersion("2.0.0");

	try{
	  $order = \Conekta\Order::create(
	    array(
	      "line_items" => array(
	        array(
	          "name" => "Donativo",
	          "unit_price" => $quantity *100, // se multipica por 100 para pesos 
	          "quantity" => 1
	        )//first line_item
	      ), //line_items
	      "shipping_lines" => array(
	        array(
	          "amount" => 0,
	           "carrier" => "None"
	        )
	      ), //shipping_lines - physical goods only
	      "shipping_contact" => array(
	        "address" => array(
	          "street1" => "none",
	          "postal_code" => "06100",
	          "country" => "MX"
	        )//address
	      ), //shipping_contact - required only for physical goods	      
	      "currency" => "MXN",
	      "customer_info" => array(
	        "customer_id" => $id
	      ), //customer_info
	      "charges" => array(
	          array(
	              "payment_method" => array(
	                      "type" => "default"
	              ) //payment_method - use customer's <code>default</code> - a card
	          ) //first charge
	      ) //charges
	    )//order
	  );
	  //print_r($order);
	} catch (\Conekta\ProcessingError $error){
		$error_final = $error->getMessage();
	   header("Location: https://trebolbit.mx/OBMex/index.html?error=".$error_final);
	  exit();
	  
	} catch (\Conekta\ParameterValidationError $error){
		$error_final = $error->getMessage();
	   header("Location: https://trebolbit.mx/OBMex/index.html?error=".$error_final);
	  exit();
	  
	} catch (\Conekta\Handler $error){
		$error_final = $error->getMessage();
	   header("Location: https://trebolbit.mx/OBMex/index.html?error=".$error_final);
	  exit();
	}
?>
<html lang="es">
<head>
	<meta charset="UTF-8">
	<title>Donativo finalizado - OBMex</title>
	<link rel="stylesheet" type="text/css" href="css/bootstrap.css"/>
	<style>
		body{
			margin: 0;
			padding: 0;
		}
		.donative{
			display: flex;
			justify-content: center;
			align-items: center;
			flex-direction: column;
			height: 100vh;
			background-color: rgba(1,1,1, .6);
			position: relative;
		}
		.title{
			color: #0338AB;
			text-align: center;
			margin: 10px 0 25px 0;
		}
		h1{
			font-size: 25px;
		}
		.title span{
			font-size: 18px;
		}
		.donative > div{
			display: flex;
			align-items: center;
			flex-direction: column;
			background-color: white;
			border-radius: 7px;
			padding: 25px 55px;
		}
		.botonEnd{
			padding: 10px 15px;
			background-color: #0338AB;
			color: white;
			border-radius: 5px;
			cursor: pointer;
		}
		.closeButtom{
			width: 100%;
			cursor: pointer;
			border-radius: 50%;
			position: relative;
		}
		.closeButtom img{
			float: right;
  			padding: 5px;
		    border-radius: 10px;
		    position: absolute;
		    right: -35px;
		    top: -10px;
		}
		.closeButtom img:hover{
			background-color: rgba(3,56,171,.6);
		}
	</style>
</head>
<body>
	<main>
		<section class="donative">
			<div>
				
				<div class="title">
					<h1>Donativo realizado con éxito</h1>
					<span>¡Gracias!</span>
				</div>
				<div class="botonEnd" onclick="cerrarModal();">
					<span>Aceptar</span>
				</div>
			</div>
		</section>
	</main>
</body>
<script type="text/javascript">
	function cerrarModal(){
		window.location.href = "https://trebolbit.mx/OBMex/index.html";
	}
</script>
</html>

Cabe mencionar que esto si funciona en un servidor y en otro no, habra algun problema con la version de php o alguna configuracion que lo restrinja ?

**Comment by tuxrick** [comment]
Y este es el error que recibo en el server que no sirve 


Parse error: syntax error, unexpected T_OBJECT_OPERATOR in /home/obmexicz/public_html/lib/lib/Conekta/ConektaResource.php on line 206

**Comment by chuckrincon** [comment]
> Cabe mencionar que esto si funciona en un servidor y en otro no, habra algun problema con la version de php o alguna configuracion que lo restrinja ?


La version de`PHP` que usa el servidor, puede estar causando el problema.

En el servidor donde funciona: ¿Que version de `PHP` esta utilizando? 
[De acuerdo con la compatibilidad de la librería](https://github.com/conekta/conekta-php/blob/master/composer.json#L18), require la version `5.3` en adelante.

**Comment by tuxrick** [comment]
Ah perfecto ! 

Checo la versión y confirmo en cuanto tenga el dato.

**Comment by tuxrick** [comment]
La versión es 5.3.29 y el error persiste.

**Comment by mauroarcet** [comment]
Hola @tuxrick, ¿Estás utilizando la versión más reciente de la librería?

La versión más reciente requiere PHP > 7

**Comment by tuxrick** [comment]
Hola ! ya probando con esa versión si funciona, Gracias !


Saludos !

Ricardo Hernandez Ramirez


2017-11-30 16:04 GMT-06:00 Mauricio Arce <notifications@github.com>:
